src/library/scala/collection/StringOps.scala

```diff
@@ -623,33 +622,63 @@ final class StringOps(private val s: String) extends AnyVal {
     *  - `LF` - line feed   (`0x0A`)
     *  - `FF` - form feed   (`0x0C`)
     */
-  def linesWithSeparators: Iterator[String] = new AbstractIterator[String] {
+  def linesWithSeparators: Iterator[String] = linesSeparated(stripped = false)
+
+  /** Lines in this string, where a line is terminated by
+   *  `"\n"`, `"\r"`, `"\r\n"`, `"\f"`, or the end of the string.
+   *  A line may be empty. Line terminators are removed.
+   */
+  def linesIterator: Iterator[String] = linesSeparated(stripped = true)
+
+  private def linesSeparated(stripped: Boolean): Iterator[String] = new AbstractIterator[String] {
+    def hasNext: Boolean = !done
+    def next(): String = if (done) Iterator.empty.next() else advance()
+
     private[this] val len = s.length
     private[this] var index = 0
-    def hasNext: Boolean = index < len
-    def next(): String = {
-      if (index >= len) Iterator.empty.next()
+    @`inline` private def done = index >= len
+    private def advance(): String = {
       val start = index
-      while (index < len && !isLineBreak(apply(index))) index += 1
-      index += 1
-      s.substring(start, index min len)
+      def result = s.substring(start, index)
+      def finish(c: Char): Unit = {
```

Comment:
Do we really need to expand 4-ish lines into 35?  Seems like we could add a couple more lines and be done with it.

Reply:
Depends on if you love the `FF` treatment and care about minor performance. Probably performance precludes splitting huge strings as opposed to streaming lines.

Reply:
No, I mean you rewrote the code in a different style which is much less compact and not obviously more performant.  The old logic is fine, except you need another couple lines to figure out whether to +1 or +2 the index if you allow multi-char separators.

Reply:
I think it has to be an improvement on `linesWithSeparators map (_.stripLineEnd)`; I think "find the EOL and optionally skip it" is OK, but I'll see about compressing the vertical part; but after I go make dinner.